### PR TITLE
Improve str layout

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR addresses the easiest requests from #1501

- [x] Fix borders of igv buttons on str_variants page
- [x] Rename "Rank" to "Index" on str_variants page
- [x] Sort str variants by Repeat Locus ID
- [x] Group 2 alleles from a single repeat together in str variants list
For clarity sake I found this solution, what do you think?
image

**How to test**:
1. Install the branch on clinical-db stage and see the difference with scout production

**Expected outcome**:
The functionalities in the checkbox should be working as described

**Review:**
- [x] code approved by dNil
- [x] tests executed by dNil

### CODE REVIEW

**Comment by dnil** [comment]
✅ button class

🤔sort and index is a good solution right now, but it would also not be difficult to fix it in Stranger esp with MIP releasing more often. It could do say any loci with pathogenic expansions first, intermediates next, larger than ref or just chromosome pos last. The latter is viable on much longer term, for more loci, up to whole genome research ExHu de novo runs. But, it’s easy enough to change back. Go for it!

**Comment by northwestwitch** [comment]
About tge sorting, it's not a good solitiin as it is yet because chromosomes are sorted like this:
1, 11, 2.. We shoukd thing about a better way to surt by chrom

**Comment by dnil** [comment]
Alphanumerical by locus might work better then?

**Comment by northwestwitch** [comment]
> Alphanumerical by locus might work better then?

Definitely!

**Comment by northwestwitch** [comment]
This is ready for review!

**Comment by dnil** [comment]
The grouping is a great start! I can review after lunch if no-one else beats me to it!

**Comment by dnil** [comment]
```
2019-12-04 13:34:03,850 - scout.server.app - ERROR: Exception on /igv [GET] [in /home/hiseq.clinical/miniconda/envs/prod-stage/lib/python3.6/site-packages/flask/app.py:1891]
Traceback (most recent call last):
 File "/home/hiseq.clinical/miniconda/envs/prod-stage/lib/python3.6/site-packages/flask/app.py", line 2446, in wsgi_app
   response = self.full_dispatch_request()
 File "/home/hiseq.clinical/miniconda/envs/prod-stage/lib/python3.6/site-packages/flask/app.py", line 1951, in full_dispatch_request
   rv = self.handle_user_exception(e)
 File "/home/hiseq.clinical/miniconda/envs/prod-stage/lib/python3.6/site-packages/flask/app.py", line 1820, in handle_user_exception
   reraise(exc_type, exc_value, tb)
 File "/home/hiseq.clinical/miniconda/envs/prod-stage/lib/python3.6/site-packages/flask/_compat.py", line 39, in reraise
   raise value
 File "/home/hiseq.clinical/miniconda/envs/prod-stage/lib/python3.6/site-packages/flask/app.py", line 1949, in full_dispatch_request
   rv = self.dispatch_request()
 File "/home/hiseq.clinical/miniconda/envs/prod-stage/lib/python3.6/site-packages/flask/app.py", line 1935, in dispatch_request
   return self.view_functions[rule.endpoint](**req.view_args)
 File "/home/hiseq.clinical/SERVER/apps/scout-stage/git/scout-stage/scout/server/blueprints/alignviewers/views.py", line 94, in igv
   if bam_files.get(counter) and bai_files.get(counter):
AttributeError: 'list' object has no attribute 'get'
```
Fails on IGV view..

File: scout/server/blueprints/variants/templates/variants/str-variants.html [hunk 1]
```diff
@@ -37,7 +37,7 @@
     <table class="table table-bordered table-hover">
       <thead>
         <tr>
-          <th style="width:8%" title="Rank position">Rank</th>
+          <th style="width:8%" title="Index">Index</th>
```
Comment (dnil) [inline scout/server/blueprints/variants/templates/variants/str-variants.html]: Good for now! As soon as Stranger 0.5.6 goes into production, we can change it back.

File: scout/server/blueprints/variants/templates/variants/str-variants.html [hunk 2]
```diff
@@ -51,7 +51,12 @@
         </tr>
       </thead>
       <tbody>
+        {% set ns = namespace(allele0='') %}
         {% for variant in variants %}
+          {% if variant.chromosome + variant.position|string != ns.allele0 %}
```
Comment (dnil) [inline scout/server/blueprints/variants/templates/variants/str-variants.html]: :+1:

File: scout/server/blueprints/variants/templates/variants/str-variants.html [hunk 3]
```diff
@@ -82,7 +87,7 @@
             <td>
               {{ variant.position }}
               {% if case.bam_files %}
-                <a class="btn btn-default btn-sm float-right" href="{{ url_for('alignviewers.igv', sample=case.sample_names, build=case.genome_build, bam=case.bam_files, bai=case.bai_files, contig=variant.chromosome, start=(variant.position - 50), stop=(variant.position + 50), center_guide="T") }}" target="_blank">IGV viewer</a>
+                <a class="btn btn-outline-secondary btn-sm float-right" href="{{ url_for('alignviewers.igv', sample=case.sample_names, build=case.genome_build, bam=case.bam_files, bai=case.bai_files, contig=variant.chromosome, start=(variant.position - 50), stop=(variant.position + 50), center_guide="T") }}" target="_blank">IGV viewer</a>
```
Comment (dnil) [inline scout/server/blueprints/variants/templates/variants/str-variants.html]: This presumably needs an update from the other variants views where checking for file existence changed.

Comment (dnil) [reply scout/server/blueprints/variants/templates/variants/str-variants.html]: (And the current master alignviewers view needs a slight change - it was perhaps not tested with trios?)

**Comment by northwestwitch** [comment]
Oh no, I'll check. Imagine if you were right when you wanted the code to check that the index files were there! :D

**Comment by dnil** [comment]
> Oh no, I'll check. Imagine if you were right when you wanted the code to check that the index files were there! :D

😆 I'd be told-you-so-ing all day! I imagine this was just because I wanted you to add the test in the first place - it would probably work just fine without..

**Comment by northwestwitch** [comment]
Now it works again! The error was not exactly there, but very close! ;)

File: scout/server/blueprints/variants/templates/variants/str-variants.html (hunk 1 above)
Comment (northwestwitch) [reply scout/server/blueprints/variants/templates/variants/str-variants.html]: Sure!

File: scout/server/blueprints/variants/templates/variants/str-variants.html (hunk 3 above)
Comment (northwestwitch) [reply scout/server/blueprints/variants/templates/variants/str-variants.html]: The error was in the alignviewers where I was checking for the existence of of the track files actually

**Comment by dnil** [comment]
Makes one long for a compiler with slightly stronger typing, doesn’t it? ;)

I think it’s just fine without the flash. 👍🏻

Is this on stage already or should I update there as well?

**Comment by northwestwitch** [comment]
I put it on stage already! I have the strong suspect that the flash wouldn't have worked anyway. The flash message is sent to the next page, but the next page is produced by igv.js and it doesn't show our flash messages..

**Comment by dnil** [comment]
Well, but then eventually they would have wandered back onto a page that could flash - and show no favicon found! 😆 Looks all good now!

**Comment by northwestwitch** [comment]
The favicon not found error will be the ruin of me

**Comment by dnil** [comment]
Hm, I actually think I fixed it locally once when I got tired of seeing it. I can see if I can't find it again, but it isn't exactly top priority..

I'll approve shortly; just want to follow up on a java script error on the page. It's probably nothing.. :wink:

**Comment by northwestwitch** [comment]
What should I fix in the end?

**Comment by dnil** [comment]
Yeah, it's the multiselect js that is feeling lonely since there is no multiselect on the page. I guess you could take it out, but that is optional.

**Comment by northwestwitch** [comment]
The multiselect error is gone now, merging as soon as the tests finish. Thanks!

**Pull request merged**